internal/controller/pkg/resolver/reconciler.go

```diff
@@ -455,7 +447,8 @@ func (r *Reconciler) findDependencyVersionToInstall(ctx context.Context, dep *v1
 		return "", errors.Wrap(err, errRewriteImage)
 	}
 	if newPath != "" {
-		ref, err = name.ParseReference(newPath, name.WithDefaultRegistry(r.registry))
+		// NOTE(phisco): it's a dependency's reference, so we can not do strict validation.
```

Comment:
worth adding a note about why this is? i can't immediately think of the obvious answer myself 😅

Reply:
updated to align with the comment above.